lib/gstreamer/transcoderbase.py

```diff
@@ -144,6 +144,8 @@ def gen_output_opts(self):
     opts += " ! queue max-size-buffers=0 max-size-bytes=0 max-size-time=0"
     if self.mode == "va_hw":
       opts += " ! vapostproc ! video/x-raw,format={format}"
+    elif self.mode == "d3d_hw":
+      opts += " ! d3d11download ! video/x-raw,format={format}"
```

Comment:
What do you mean by "encode configure"? This is the output for the second channel linked with checksumsink2 (dump yuv from decoder output), because decoder output is d3d11 memory, we need use d3d11download here to get "video/x-raw" caps which can linked with checksumsink2.

Reply:
This trancoding case is about surface implicit mapping between d3d and msdk, i.e., d3d11dec +msdkenc. So we stick to msdkenc here.
The complete cmd:
<pre>gst-launch-1.0 -vf filesrc location=input.264 ! h264parse ! d3d11h264dec ! tee name=transcoder ! queue max-size-buffers=0 max-size-bytes=0 max-size-time=0 ! msdkh264enc ! video/x-h264,profile=main ! h264parse ! filesink location=output.h264 transcoder. ! queue max-size-buffers=0 max-size-bytes=0 max-size-time=0 ! d3d11download ! video/x-raw,format=NV12 ! checksumsink2 file-checksum=false qos=false eos-after=71 frame-checksum=false plane-checksum=false dump-output=true dump-location=output.yuv

Reply:
There is no need to add anything extra in encoder part for "gen_output_opts" (the mode in encoder part is ["sw", "hw", "lp"]) , we just need to choose "hw" for msdkenc in specific test case (see in https://github.com/intel-innersource/applications.media.middleware.validation.vaapi-fits-smoke/pull/296/files#diff-aab09a6960e8975e143d35c4381dcd27176160094ea3af8788859b447f4db8de).